### DISCUSSION TITLE
DataGrid edited cell background and border

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#11022] [Q&A] [answered]
I have a DataGrid with an editable column.
By default, it is rather hard to see when a cell is selected for editing.

How can I change the appearance of the cell being edited (background color and/or border)?

**Comment by maxkatz6** [comment]
You can write a style with selector "DataGridCell:current /template/ Rectangle#CurrencyVisual" and change background in the selector. Or even write a selector to the TextBox directly "DataGridCell:current TextBox".
Based on https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L144
Or https://github.com/AvaloniaUI/Avalonia/blob/0.10.19/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L153 (if stable branch)

**Reply by wgraaf** [reply]
Thanks!
I got this to work:

    <Style Selector="DataGridCell:current">
      <Setter Property="Background" Value="Yellow"/>
      <Setter Property="Foreground" Value="Black"/>
    </Style>

I could not get any result with 

    <Style Selector="DataGridCell:current /template/ Rectangle#CurrencyVisual">...

or

    <Style Selector="DataGridCell:current TextBox">...

**Reply by wgraaf** [reply]
Also, I do not seem to be able to set the border color and thickness:

    <Style Selector="DataGridCell:current">
      <Setter Property="BorderThickness" Value="2"/>
      <Setter Property="BorderBrush" Value="Black"/>
    </Style>

Does not work.
How would I go about putting a thick border around the edited cell?

**Reply by timunie** [reply]
Check the original styles linked above and probably also use DevTools ([F12] while debugging) to check the visual tree. 

Note: you cannot override locally set values. So you maybe need to override the entire ControlTemplate, see: https://github.com/AvaloniaUI/Avalonia/blob/45c009e3d2d7e1f383dfff0cd33b95de0b4b11a1/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L220-L237

**Comment by wgraaf** [comment, marked as answer]
I ended up using a focus:within selector to highlight the appearance of the cell being edited:

    <Style Selector="DataGridCell:focus-within">
      <Setter Property="Background" Value="Yellow"/>
      <Setter Property="Foreground" Value="Black"/>
    </Style>

**Reply by nasterhe** [reply]
@wgraaf Is there a way to set the background of an arbitrary DataGridCell based on the binded value? Thank you.
